nixos/modules/services/networking/go-autoconfig.nix

```diff
@@ -0,0 +1,98 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.go-autoconfig;
+  configFile = cfg: pkgs.writeText "go-autoconfig.conf" ''
+    service_addr: ":${cfg.port}"
+
+    domain: ${cfg.domain}
+
+    imap:
+      server: ${cfg.imap.server}
+      port: ${cfg.imap.port}
+
+    smtp:
+      server: ${cfg.smtp.server}
+      port: ${cfg.smtp.port}
+  '';
+
+in {
+  options = {
+    services.go-autoconfig = {
+
+      enable = mkEnableOption "IMAP/SMTP autodiscover feature for mail clients";
+
+      port = mkOption {
+        type = types.port;
+        default = 1323;
+        description = mdDoc "Port to listen on (use 0 for random port).";
+      };
+
+      domain = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = mdDoc "Domainname used for autoconfig server.";
+        example = "autoconfig.example.org";
+      };
+
+      imap = {
+
+        server = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = mdDoc "Domainname used for autoconfig server.";
+          example = "imap.example.org";
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 993;
+          description = mdDoc "IMAP port your mail server is listening on.";
+        };
+
+      };
+
+      smtp = {
+
+        server = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = mdDoc "Domainname used for autoconfig server.";
+          example = "smtp.example.org";
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 587;
+          description = mdDoc "SMTP port your mail server is listening on.";
+        };
+
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd = {
+      services.go-autoconfig = {
+        wantedBy = [ "multi-user.target" ];
+        description = "IMAP/SMTP autodiscover server";
+        after = [ "network.target" ];
+        restartTriggers = [ configFile ];
```

Comment:
`configFile` will be a store path, so this is pretty much implied by being used elsewhere in the service definition. restart triggers are mostly necessary when the referenced file is somewhere in `/etc` or doesn't change its path on system rebuilds for other reasons